Assets/Scripts/Lib/Cache.cs

```diff
@@ -41,7 +41,7 @@ public void Set(string key, T item)
         public T Get(string key)
         {
             CacheObject<T> cacheObject = new CacheObject<T>();
-            if(cache.TryGetValue(key, out cacheObject))
+            if(key != null && cache.TryGetValue(key, out cacheObject))
```

Comment:
No need to construct the object if key is null. Possibly also investigate if `new CacheObject()` needs to be called here at all.